File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        # Create a new linked list to store merged list
        newHead = dummyHead = ListNode()
        # While list1 and list2 exist
        while list1 and list2:
            if list1.val < list2.val:
                dummyHead.next = list1
                list1 = list1.next
            else:
                dummyHead.next = list2
                list2 = list2.next
            dummyHead = dummyHead.next
        
        # if only 1 list exists
        if list1:
            dummyHead.next = list1
        if list2:
            dummyHead.next = list2
        # Return new head of list
        return newHead.next
        

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
        // Check if list is empty or at the end
    	if(l1 == null) 
        {return l2}
		if(l2 == null) 
        {return l1}
        // Use recursion, insert lower value into  list
		if(l1.val < l2.val){
			l1.next = mergeTwoLists(l1.next, l2)
			return l1
		} else{
			l2.next = mergeTwoLists(l1, l2.next)
			return l2
		}
};